src/Pessoa.php

```diff
@@ -877,52 +879,30 @@ public static function listarColegiados()
         return DB::fetchAll($query, $param);
     }
 
-    /**
-     * Método que lista membros titulares para um dado colegiado
-     * 
-     * @param Integer $codclg código do colegiado pode ser obtido com listarColegiados()
-     * @return Array lista de membros do colegiado selecioando
-     * @author @thiagogomesverissimo - 23/11/2021
-     *
-     */
-    public static function listarTitularesDoColegiado(int $codclg)
-    {
-        $query = DB::getQuery('Pessoa.listarTitularesDoColegiado.sql');
-
-        $unidades = getenv('REPLICADO_CODUNDCLG');
-        $query = str_replace('__unidades__', $unidades, $query);
-
-        $dtafimmdt = Date('Y-m-d') . ' 00:00:00';
-
-        $param = [
-            'dtafimmdt' => $dtafimmdt,
-            'codclg'    => $codclg
-        ];
-
-        return DB::fetchAll($query, $param);
-    }
-
     /**
      * Método que lista membros titulares e suplentes para um dado colegiado
      * 
      * @param Integer $codclg código do colegiado pode ser obtido com listarColegiados()
+     * @param String $sglclg Alguns colegiados tem o mesmo código e podem ser diferenciados através da sigla, por esta razão é recomendado buscar através do par código e sigla do colegiado.
      * @return Array lista de membros do colegiado selecioando
      * @author @thiagogomesverissimo - 23/11/2021
      *
      */
-    public static function listarTitularesSuplentesDoColegiado(int $codclg)
+    public static function listarTitularesSuplentesDoColegiado(int $codclg, string $sglclg = '')
     {
         $query = DB::getQuery('Pessoa.listarTitularesSuplentesDoColegiado.sql');
 
         $unidades = getenv('REPLICADO_CODUNDCLG');
         $query = str_replace('__unidades__', $unidades, $query);
+        $query = str_replace('__sglclg__', "AND PARTICIPANTECOLEG.sglclg = '$sglclg'" , $query);
```

Comment:
o $sglclg não deveria passar no $param? assim não precisaria do replace.

Reply:
Não passei pelo $param porque é um parâmetro opcional e que se repete na query.
Deste modo se no .sql eu colocasse "AND PARTICIPANTECOLEG.sglclg = :sglclg", e se fosse vazio eu teria que apagar esta linha, né?


Não dá para adicionar um if (para verificar se não é vazio) e colocar a parte do "AND PARTICIPANTECOLEG.sglclg = :sglclg" no final, pq esta condição se repete e no fim da query tem um order by...